presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AnySymbol.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.assertions;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.tree.SymbolReference;
+
+class AnySymbol extends Symbol
+        implements PlanTestSymbol
+{
+    AnySymbol()
+    {
+        super("*");
+    }
+
+    @Override
+    public Symbol toSymbol(ExpressionAliases aliases)
+    {
+        return this;
+    }
+
+    @Override
+    public SymbolReference toSymbolReference()
+    {
+        return new AnySymbolReference();
+    }
+
+    @Override
+    public int hashCode()
+    {
+        /*
+         * If you find a use case for putting AnySymbol in a hash table, feel free to
+         * implement this.
+         */
+        throw new UnsupportedOperationException("Test object");
```

Comment:
Added a comment explaining why an implementation that satisfies the general contract is impossible, and why it's unlikely to be useful to implement a method that intentionally violates the contract.